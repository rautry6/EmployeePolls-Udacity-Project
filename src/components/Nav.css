.link {
    color: #617dfb;
    word-spacing: 4ch;
    text-decoration: none;
}

.nav-avatar{
    width: 40px;
    height: 40px;
    border-radius: 50%;
    margin-right: 10px;
}

.authed-user{
    font-size: 1rem;
    font-weight: bold;
    color: rgb(0, 0, 0);
}