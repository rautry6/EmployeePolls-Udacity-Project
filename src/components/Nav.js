import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { setAuthedUser } from "../actions/authedUser.js";
import { useState, useEffect } from "react";
import "./Nav.css";

const Nav = (props) => {
  const [authedUser, setAuthUser] = useState(props.authedUser);
  let users = props.users;

  useEffect(() => {
    setAuthUser(props.authedUser);
  }, [props.authedUser]);

  const onLogout = (e) => {
    setAuthUser("");
    props.dispatch(setAuthedUser(null));
  };

  return (
    <div className="nav">
      <Link to="/homepage" className="link">Home </Link>

      <Link to="/leaderboard" className="link">Leaderboard </Link>

      <Link to="/new" className="link">New </Link>

      <Link to="/answered" className="link">Answered </Link>

      <Link to="/add" className="link">Add Poll </Link>

      {authedUser !== null ? (
        <span>
          <span className="authed-user">{authedUser}</span>
          <img
            src={users[authedUser].avatarURL}
            alt="avatar"
            className="nav-avatar"
          />
        </span>
      ) : (
        <span></span>
      )}
      <Link to="/" onClick={onLogout} className="link">
        Logout
      </Link>
    </div>
  );
};

function mapStateToProps(state) {
  return {
    authedUser: state.authedUser,
    users: state.users,
  };
}

export default connect(mapStateToProps)(Nav);
