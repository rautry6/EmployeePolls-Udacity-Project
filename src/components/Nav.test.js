import { render, fireEvent } from "@testing-library/react";
import Nav from "./Nav";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../reducers";
import middleware from "../middleware";
import { BrowserRouter } from "react-router-dom";

const store = createStore(rootReducer, middleware);

describe("Nav", () => {
  it("will match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
