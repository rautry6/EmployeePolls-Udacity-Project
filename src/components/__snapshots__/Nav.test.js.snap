// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav will match snapshot 1`] = `
<div>
  <div
    class="nav"
  >
    <a
      href="/homepage"
    >
      Home 
    </a>
    <a
      href="/leaderboard"
    >
      Leaderboard 
    </a>
    <a
      href="/new"
    >
      New 
    </a>
    <a
      href="/answered"
    >
      Answered 
    </a>
    <a
      href="/add"
    >
      Add Poll 
    </a>
    <span />
    <a
      href="/"
    >
      Logout
    </a>
  </div>
</div>
`;
